symphony-ws/src/main/java/se/havochvatten/symphony/service/MetaDataService.java

```diff
@@ -43,37 +43,37 @@ public class MetaDataService {
             "AND m2.metaField IN ('title', 'symphonytheme')" +
             "AND m2.language = :language)))";
 
-    public MetadataDto findMetadata(String baselineName, String preferredLanguage, boolean sparse) throws SymphonyStandardAppException {
+    public MetadataDto findMetadata
+        (String baselineName, String preferredLanguage, boolean sparse) throws SymphonyStandardAppException {
         BaselineVersion baseline = baselineVersionService.getVersionByName(baselineName);
         MetadataDto metadataDto = new MetadataDto();
         metadataDto.setEcoComponent(getComponentDto("Ecosystem", baseline.getId(), preferredLanguage, sparse));
         metadataDto.setPressureComponent(getComponentDto("Pressure", baseline.getId(), preferredLanguage, sparse));
         metadataDto.setLanguage(preferredLanguage);
         return metadataDto;
-
     }
 
-    public Map<Integer, String>
-        getComponentTitles(int baselineVersionId, LayerType category, String preferredLanguage)
-        throws SymphonyStandardAppException {
+    public Map<Integer, String> getSingleMetaFieldForComponent
+        (int baselineVersionId, LayerType category, String field, String preferredLanguage) {
 
         TypedQuery<Tuple> bandTitlesQuery = em.createQuery(
             "SELECT m.band.bandnumber, m.metaValue FROM Metadata m " +
                     "WHERE m.band.baseline.id = :baselineVersionId " +
-                    "AND m.metaField = 'title' " +
+                    "AND m.metaField = :field " +
                     "AND m.band.category = :category " +
                     "AND (m.language = :language " +
                         "OR (m.language = m.band.baseline.locale " +
                         "AND m.metaField NOT IN " +
                             "(SELECT m2.metaField FROM Metadata m2 " +
                             "WHERE m2.band.baseline.id = :baselineVersionId " +
-                            "AND m2.metaField = 'title' " +
+                            "AND m2.metaField = :field " +
                             "AND m2.language = :language)))" +
                     "ORDER BY m.band.bandnumber", Tuple.class);
 
         return bandTitlesQuery
             .setParameter("baselineVersionId", baselineVersionId)
             .setParameter("category", category == LayerType.ECOSYSTEM ? "Ecosystem" : "Pressure")
+            .setParameter("field", field)
```

Comment:
Should we only allow a set of values here?

Reply:
As per our discussion, I think it's preferable to keep the method versatile.
The method isn't exposed to arbitrary input at present, and in principle there's no good argument for 'hardcoding' certain metadata field names anywhere.
This method is currently called from within the application by the constant string values "title" and "symphonytheme" only. If it is to be exposed to user input in the future it would be better to validate it client side, but it'd be pertinent to throw and catch empty results as exceptions.

Anyway, I appreciate that you made me look in this direction, it brought my attention to that the variable naming here bandTitlesQuery is inadequate after the change, let's fix that